ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -292,15 +293,15 @@ public CloudBlobMetadata getBlobMetadata(BlobId blobId) throws BlobStorageExcept
   /**
    * Update the metadata for the specified blob.
    * @param blobId The {@link BlobId} to update.
-   * @param fieldName The metadata field to modify.
-   * @param value The new value.
+   * @param updateFields Map of field names and new values to modify.
    * @return a {@link UpdateResponse} with the updated metadata.
    * @throws BlobStorageException if the blob does not exist or an error occurred.
    * @throws IllegalStateException on request timeout.
    */
-  public UpdateResponse updateBlobMetadata(BlobId blobId, String fieldName, Object value) throws BlobStorageException {
+  public UpdateResponse updateBlobMetadata(BlobId blobId, Map<String, Object> updateFields)
+      throws BlobStorageException {
     Objects.requireNonNull(blobId, "BlobId cannot be null");
-    Objects.requireNonNull(fieldName, "Field name cannot be null");
+    updateFields.keySet().stream().forEach(field -> Objects.requireNonNull(updateFields.get(field)));
```

Comment:
nit: can you add the field name as the second parameter to requireNoNull method.

Reply:
done